app/models/censuses/census_report.rb

```diff
@@ -0,0 +1,286 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Censuses
+  class CensusReport
+    include ArelHelper
+    include Filter::ControlSections
+
+    def initialize(filter)
+      @filter = filter
+    end
+
+    # what projects should be included?
+    def census_projects_scope
+      scope = GrdaWarehouse::Hud::Project.residential.
+        viewable_by(@filter.user).
+        where(id: @filter.effective_project_ids)
+
+      # Limit ES projects to Night-by-night only by filtering out Entry/Exit ES projects
+      scope = scope.where.not(ProjectType: 1, TrackingMethod: 0) if @filter.limit_es_to_nbn
+      scope
+    end
+
+    # what data should be included?
+    def census_data_scope(project_scope)
+      GrdaWarehouse::Census::ByProject.joins(:project).merge(project_scope)
+    end
+
+    # JSON Shape:
+    # { datasource_id: {
+    #     organization_id: { // include "all" for totals if more than one project_id
+    #       project_id: { // include "all" for totals if more than one project_id
+    #         "datasets": [
+    #           {
+    #             "label": "Client Count",
+    #             "data": [
+    #               {
+    #                 "x": date,
+    #                 "y": value,
+    #                 "yesterday": value // To allow change computation for the datum
+    #               }
+    #             ]
+    #           },
+    #           { "label": "Bed Inventory Count"
+    #             "data": [
+    #               {
+    #                 "x": date,
+    #                 "y": value
+    #               }
+    #             ]
+    #           }
+    #         ],
+    #         "title": {
+    #           "display": true,
+    #           "text": string
+    #         }
+    #       }
+    #     }
+    #   }
+    # }
+
+    def for_date_range
+      user = @filter.user
+      start_date = @filter.start
+      end_date = @filter.end
+
+      @shape ||= {}
+      project_scope = census_projects_scope
+
+      case @filter.aggregation_level.to_sym
+      when :by_project
+        project_scope.joins(:organization, :data_source).each do |project|
+          for_project(start_date, end_date, project, user: user)
+        end
+      when :by_organization
+        organizations = GrdaWarehouse::Hud::Organization.joins(:projects).merge(project_scope).distinct
+        organizations.each do |organization|
+          for_organization(start_date, end_date, organization, project_scope, user: user)
+        end
+      when :by_data_source
+        data_sources = GrdaWarehouse::DataSource.joins(:projects).merge(project_scope).distinct
+        data_sources.each do |ds|
+          for_data_source(start_date, end_date, ds, project_scope)
+        end
+      when :by_project_type
+        [:ph, :th, :es, :so, :sh].each do |project_type|
```

Comment:
I wonder if we should do:
```suggestion
       GrdaWarehouse::Hud::Project::RESIDENTIAL_PROJECT_TYPES.keys.each do |project_type|
```
I think that'll result in some additional charts, and some duplication for PH, but I bet people would like the RRH/PSH/OPH breakouts.

Reply:
Hmm okay. `GrdaWarehouse::Hud::Project::RESIDENTIAL_PROJECT_TYPES.keys` is `[:ph, :rrh, :psh, :oph, :th, :es, :so, :sh, :permanent_housing, :transitional_housing, :emergency_shelter, :street_outreach, :safe_haven]`, which we don't want because it will duplicate a bunch.

I think the list we want is `[:ph, :es, :th, :sh, :so, :rrh, :psh, :oph]` - is that correct? if so I can get it from `GrdaWarehouse::Hud::Project::RESIDENTIAL_TYPE_TITLES.keys`.

Reply:
Yes, good catch.

Reply:
Updated with https://github.com/greenriver/hmis-warehouse/pull/2628/commits/c062f34b0d16468dca610857070af3d7317ff1d8

example:

![hmis-warehouse dev test_censuses_utf8=%E2%9C%93 filters%5Bstart%5D=Oct+28%2C+2021 filters%5Bend%5D=Nov+27%2C+2021 filters%5Bcoc_codes%5D%5B%5D= filters%5Bproject_ids%5D%5B%5D= filters%5Bdata_source_ids%5D%5B%5D= filters%5Bproject_type_codes](https://user-images.githubusercontent.com/5333982/204366617-70b37b6d-630f-4b84-8910-ed6f58a9cd8c.png)